src/types/RequireDefined.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * Makes the specified optional properties required. Also works where the property is
+ * defined as `property: foo | undefined`, ensuring that `property` is not undefined. For
+ * example:
+ *
+ * ```
+ * interface MyObject {
+ *    prop: string;
+ *    optionalProp?: number;
+ *    propWithUndefined?: string | undefined;
```

Comment:
Did you mean to put a `?` on this property? The description above talks about `property: foo | undefined` but it's basically `property?: foo | undefined` here. That works too, but just checking.